app/javascript/controllers/media_tag_controller.js

```diff
@@ -50,7 +50,11 @@ export default class extends Controller {
                  fileLabel: dataset.fileLabel || '' }
       })
 
-    window.dispatchEvent(new CustomEvent('thumbnails-found', { detail: thumbnails }))
+    // Timeout is set because when the page is cached, the event fires before the content_list_controller is mounted.
```

Comment:
Is `findThumbnails` idempotent? Could you dispatch a custom event in `content_list_controller#connect` and listen for it in `media_tag_controller` and run `findThumbnails` again? If you've been down this road already, I certainly won't stand in the way of progress.

Reply:
I have. It ends up being stuck in an endless loop. Also sometimes it doesn't work because the media_tag_controller loads after the content_list_controller and so it gets really funky.

Reply:
I believe it. Thanks!